src/games/SpaceInvaders/SpaceInvaders.jsx

```diff
@@ -0,0 +1,251 @@
+import GameBoard from "./GameBoard.jsx";
+import CONFIG, { PLAYER_HEIGHT, PLAYER_WIDTH } from "./config";
+import movePlayer from "./movePlayer.js";
+import handleShots from "./moveShot.jsx";
+import handleEnemies from "./handleEnemies.jsx";
+import spawnEnemy from "./spawnEnemies.jsx";
+import readOutLoud from "./sounds.jsx";
+
+export default class SpaceInvaders extends GameComponent {
+  constructor(props) {
+    super(props);
+    var myUserId = this.getMyUserId();
+    this.state = {
+      myId: myUserId,
+      enemies: spawnEnemy(5),
+      status: "menu",
+      playerOne: {
+        left: 0,
+        top: 400,
+        health: 3,
+        shots: []
+      },
+      playerTwo: {
+        left: 300,
+        top: 400,
+        health: 3,
+        shots: []
+      },
+      score: 0
+    };
+
+    this.getSessionDatabaseRef().set({
+      playerNames: UserApi.getName(this.getSessionUserIds()),
+      //shourld I call the spawnenemy function here instead or refer to state?
+      enemies: this.state.enemies,
+      status: this.state.status,
+      playerOne: this.state.playerOne,
+      playerTwo: this.state.playerTwo,
+      score: this.state.score
+    });
+
+    this.isCreator = () => {
+      if (this.getSessionCreatorUserId() === this.getMyUserId()) {
+        return "Creator";
+      } else {
+        return false;
+      }
+    };
+
+    this.timer = null;
+  }
+
+  //this.getSessionDatabaseRef().update({});
+
+  onSessionDataChanged(data) {
+    this.setState({
+      myId: this.state.myId,
+      enemies: data.enemies,
+      status: data.status,
+      playerOne: data.playerOne,
+      playerTwo: data.playerTwo,
+      score: data.score
+    });
+    // console.log(data);
+  }
+
+  playing = () => {
+    // console.log(this.state, "playing state");
+    this.getSessionDatabaseRef().update({
+      status: "playing"
+    });
+    this.timer = setInterval(() => {
+      if (this.state.status === "playing") {
+        // console.log("playing, game is running");
+        this.enemyFall(10);
+        this.shotMovement();
+      }
+    }, CONFIG.FRAMES_PER_SECOND);
+  };
+
+  shotMovement = () => {
+    let enemies = this.state.enemies || [];
+    if (this.isCreator()) {
+      let shots = this.state.playerOne.shots || [];
+      let newShots = handleShots(shots, enemies);
+      let newEnemies = handleEnemies(shots, enemies);
+      this.getSessionDatabaseRef().update({
+        playerOne: {
+          ...this.state.playerOne,
+          shots: newShots
+        },
+        enemies: newEnemies
+      });
+    } else {
+      let shots = this.state.playerTwo.shots || [];
+      let newShots = handleShots(shots, enemies);
+      let newEnemies = handleEnemies(shots, enemies);
+      this.getSessionDatabaseRef().update({
+        playerTwo: {
+          ...this.state.playerTwo,
+          shots: newShots
+        },
+        enemies: newEnemies
+      });
+    }
+  };
+
+  enemyFall = num => {
+    // if (this.isCreator()) {

```

Comment:
Why did you remove this `isCreator()` check? This means both player one and two send the same `newEnemies` list to Firebase, which might lead to problems. I think it makes sense for only the game creator to send this data to Firebase.